### ISSUE TITLE
Cookiestore?

### ISSUE THREAD

**Issue by tipsy** [spinscale/javalin-cookie-session-store#1]
I saw your project on Twitter. Just a quick question, did you see the `ctx.cookieStore()` methods?

```
ctx.cookieStore(key)                    // get cookie store value
ctx.cookieStore(key, value)             // set a cookie store value
ctx.clearCookieStore()                  // clear the cookie store
```

Maybe we could expand it to include a secret too? 🤔

**Comment by spinscale** [comment]
yeah, saw them, but dismissed them due to the lack of secrets :-)

I like the idea of having a configurable option of making the cookiestore encrypted (potentially in the config of a Javalin app) by setting a secret. Or maybe even simpler, have two different implementations of a `CookieStore`, one encrypting the cookies and specify those in the app config.

What I didn't like from an implementation perspective was the serialization after setting a value. This seems expensive to me when using encryption, therefore this plugin is only doing it once in the write handler.

Maybe just allow for a pluggable CookieStore (this would prevent dragging additional dependencies in the core like the JWT ones) with an option for lazy serialization (not sure how this could happen looking at the code at 11pm :-)

P.S. I haven't said it, but thanks for Javalin! If you happen to visit Munich, first rounds of beer are on me.

**Comment by tipsy** [comment]
I'm open to both approaches. This seems like something you could configure with a functional interface, like the JSON mapping in Javalin.

I'll make an issue over on the Javalin tracker. Would you like to work on it?

> P.S. I haven't said it, but thanks for Javalin! If you happen to visit Munich, first rounds of beer are on me.

I'm happy you like it!

**Comment by spinscale** [comment]
so my kotlin skills are mediocre at best, but I'll try to take a look at it some time soon I hope. I'd opt for the first approach of being able to specify an cookiestore implementation when creating an App, and using the default CookieStore unless another implementation is specified. Plus adding a method for late serialization
